src/components/panels/details/info-multiple-selection.tsx

```diff
@@ -1,46 +1,127 @@
+
+    const hardRefresh = React.useCallback(() => {
+        dispatch(setRefreshNodes());
+        dispatch(setRefreshEdges());
+        dispatch(saveGraph(graph.current.export()));
+    }, [dispatch, setRefreshNodes, setRefreshEdges, saveGraph]);
     const graph = React.useRef(window.graph);
 
-    const handleChange = (node: string) => {
-        dispatch(clearSelected());
-        dispatch(addSelected(node));
+    const handleChange = (id: string) => {
+        dispatch(setSelected([id]));
+    };
+
+    const getName = (id: string) => {
+        if (id.startsWith('stn') || id.startsWith('misc_node')) {
+            const attr = graph.current.getNodeAttributes(id);
+            const type = attr.type;
+            return id.startsWith('stn') ? (attr[type] as StationAttributes).names.join('/') : type;
+        } else if (id.startsWith('line')) {
+            const [s, t] = graph.current.extremities(id);
+            const source = graph.current.getSourceAttributes(id);
+            const target = graph.current.getTargetAttributes(id);
+            const sT = source.type;
+            const tT = target.type;
+            return (
+                (s.startsWith('stn') ? (source[sT] as StationAttributes).names[0] : sT) +
+                ' - ' +
+                (t.startsWith('stn') ? (target[tT] as StationAttributes).names[0] : tT)
+            );
+        }
+    };
+
+    const handleChangeColor = (color: Theme) => {
+        selected.forEach(id => {
+            if (graph.current.hasEdge(id)) {
+                const thisType = graph.current.getEdgeAttributes(id).style;
+                const attrs = graph.current.getEdgeAttribute.bind(graph.current)(id, thisType);
+                if (thisType !== LineStyleType.River && (attrs as AttributesWithColor)['color'] !== undefined) {
+                    (attrs as AttributesWithColor)['color'] = color;
+                }
+                graph.current.mergeEdgeAttributes.bind(graph.current)(id, { [thisType]: attrs });
+            } else if (graph.current.hasNode(id)) {
+                const thisType = graph.current.getNodeAttributes(id).type;
+                const attrs = graph.current.getNodeAttribute.bind(graph.current)(id, thisType);
+                if ((attrs as AttributesWithColor)['color'] !== undefined)
+                    (attrs as AttributesWithColor)['color'] = color;
+                graph.current.mergeNodeAttributes.bind(graph.current)(id, { [thisType]: attrs });
+            }
+        });
+        hardRefresh();
     };
 
+    const [isThemeRequested, setIsThemeRequested] = React.useState(false);
+    React.useEffect(() => {
+        if (isThemeRequested && output) {
+            handleChangeColor(output);
+            setIsThemeRequested(false);
+        }
+    }, [output?.toString()]);
+
     return (
         <Box>
             <Heading as="h5" size="sm">
                 {t('panel.details.selected')} {selected.length}
             </Heading>
+            <RmgFields
```

Comment:
Do we really want a `RmgFields` here? `<RmgLabel label={t('...')}><ThemeButton .../></RmgLabel>` might be simplier.

Reply:
Also, should we hide this if all the selected items can not have a color change? E.g. all of them are rivers.